wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -218,7 +197,16 @@ class WirePlugin : Plugin<Project> {
       val task = project.tasks.register(taskName, WireTask::class.java) { task: WireTask ->
         task.group = GROUP
         task.description = "Generate protobuf implementation for ${source.name}"
-        task.source(protoSourceInput.configuration)
+
+        // Flatten all the input files here. Changes to any of them will cause the task to re-run.
```

Comment:
Yes!

Reply:
~Is it even required? Given there is `sourceInput` and `protoInput` below, might be possible for `WireTask` to extend `DefaultTask` instead of `DefaultTask`?~

Edit I read [below](https://github.com/square/wire/pull/2838/files#r1503972524) we want `NO-SOURCE` if no source is found. In which case, I think `protoSource` should be in source and `protoPath` passed out of the band?

Reply:
I think you’re strictly correct, and also I don’t think it’ll matter much in practice?
Both skipping the task and running it when there’s no sources will both be super fast and rare.

Reply:
Agreed 👍